File: frontend/src/lib/telecare.ts
import { apiRequest } from "@/lib/api";

export type TelecareProvider = {
  id: string;
  full_name: string;
  role: "PSYCHOLOGIST" | "CLINIC" | "HOSPITAL";
  session_price_brl: number;
  platform_fee_percent: number;
  platform_fee_brl: number;
  provider_payout_brl: number;
  accepts_telecare: boolean;
};

export type TelecareSession = {
  id: string;
  requester_user_id: string;
  requester_name: string;
  provider_user_id: string;
  provider_name: string;
  provider_role: string;
  status: "REQUESTED" | "ACCEPTED" | "IN_SESSION" | "COMPLETED" | "CANCELED";
  room_code: string;
  session_price_brl: number;
  platform_fee_percent: number;
  platform_fee_brl: number;
  provider_payout_brl: number;
  payment_status: string;
  notes: string | null;
  requested_at: string;
  scheduled_for: string | null;
  updated_at: string;
  room_title: string;
  video_engine_status: string;
};

export type TelecareJoin = {
  session_id: string;
  room_code: string;
  provider_name: string;
  requester_name: string;
  join_url: string;
  expires_at: string;
  video_engine: "DAILY";
};

export async function listTelecareProviders(): Promise<TelecareProvider[]> {
  return apiRequest<TelecareProvider[]>("/telecare/providers");
}

export async function listTelecareSessions(): Promise<TelecareSession[]> {
  return apiRequest<TelecareSession[]>("/telecare/sessions");
}

export async function requestTelecareSession(input: {
  provider_user_id: string;
  notes?: string;
}): Promise<TelecareSession> {
  return apiRequest<TelecareSession>("/telecare/sessions", {
    method: "POST",
    body: JSON.stringify({
      provider_user_id: input.provider_user_id,
      notes: input.notes?.trim() || null
    })
  });
}

export async function updateTelecareSessionStatus(input: {
  sessionId: string;
  status: TelecareSession["status"];
}): Promise<TelecareSession> {
  return apiRequest<TelecareSession>(`/telecare/sessions/${input.sessionId}/status`, {
    method: "PATCH",
    body: JSON.stringify({ status: input.status })
  });
}

export async function joinTelecareSession(sessionId: string): Promise<TelecareJoin> {
  return apiRequest<TelecareJoin>(`/telecare/sessions/${sessionId}/join`, {
    method: "POST"
  });
}

File: backend/app/schemas/telecare.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class TelecareProviderResponse(BaseModel):
    id: str
    full_name: str
    role: str
    session_price_brl: float
    platform_fee_percent: float
    platform_fee_brl: float
    provider_payout_brl: float
    accepts_telecare: bool


class TelecareSessionCreate(BaseModel):
    provider_user_id: str
    notes: str | None = Field(default=None, max_length=1000)
    scheduled_for: datetime | None = None


class TelecareSessionStatusUpdate(BaseModel):
    status: Literal["ACCEPTED", "IN_SESSION", "COMPLETED", "CANCELED"]


class TelecareSessionResponse(BaseModel):
    id: str
    requester_user_id: str
    requester_name: str
    provider_user_id: str
    provider_name: str
    provider_role: str
    status: str
    room_code: str
    session_price_brl: float
    platform_fee_percent: float
    platform_fee_brl: float
    provider_payout_brl: float
    payment_status: str
    notes: str | None
    requested_at: datetime
    scheduled_for: datetime | None
    updated_at: datetime
    room_title: str = "Sala ZETTA de teleatendimento"
    video_engine_status: str = "DAILY_NOT_CONFIGURED"


class TelecareJoinResponse(BaseModel):
    session_id: str
    room_code: str
    provider_name: str
    requester_name: str
    join_url: str
    expires_at: datetime
    video_engine: str = "DAILY"
